### PR TITLE
docs(ref): Highlight commands to answer dep resolution questions

### LINKED ISSUE #6666: Add `cargo lucifer` subcommand
Add a new subcommand that brings light (get it?) to cargo's dependency resolving, answering questions like

- Why was that dependency included?
- Why was that feature on this dependency enabled?
- Which other versions were considered and why was this one chosen?

- - -

Yes, I made this issue because I saw the last issue was number 6665. Feel free to close it as "bad joke".

### PR DESCRIPTION
**Pull request by epage**
Previously, in the #6666, the aptly named `cargo lucifer` was suggested for answering resolver questions.

Turns out most of these can be answered now, between logging and `cargo tree`.
This adds troubleshooting tips to help highlight the use of these commands to answer these questions.

Fixes #6666

### CODE REVIEW

**Review by ehuss — APPROVED** [review]
👍

**Review by weihanglo — APPROVED** [review]
I wonder if we might want to put all troubleshootings in one place. Maybe under [FAQ](https://doc.rust-lang.org/nightly/cargo/faq.html)? It has aleady got some lengthy contents like [Why is Cargo rebuilding my code?](https://doc.rust-lang.org/nightly/cargo/faq.html#why-is-cargo-rebuilding-my-code) and [What does “version conflict” mean and how to resolve it?](https://doc.rust-lang.org/nightly/cargo/faq.html#what-does-version-conflict-mean-and-how-to-resolve-it)

* https://doc.rust-lang.org/nightly/cargo/reference/resolver.html#troubleshooting
* https://doc.rust-lang.org/nightly/cargo/reference/publishing.html#troubleshooting-github-team-access-errors

That said, I do feel this PR fits in the natural reading style.

File: src/doc/src/reference/resolver.md [hunk 1]
```diff
@@ -517,6 +551,17 @@ But, if you run into this situation, the [`cargo update`] command with the
 
 [`cargo update`]: ../commands/cargo-update.md
 
+### Why wasn't a newer version selected?
```
Comment (weihanglo) [inline src/doc/src/reference/resolver.md:554]: nit: we can link to an issue having some guidance for debugging as an example. I can't recall any specific one at this moment but I believe there are some.

Comment (epage) [reply src/doc/src/reference/resolver.md:554]: If you have one, that'd be great.  I can't think of ay.

**Comment by weihanglo** [comment]
Go ahead and merge this!
@bors r+

**Pull request merged**